types/node/index.d.ts

```diff
@@ -5755,57 +5755,77 @@ declare module "crypto" {
 
     export interface Hash extends NodeJS.ReadWriteStream {
         update(data: string | ArrayBufferView): Hash;
-        update(data: string | ArrayBufferView, input_encoding: Utf8AsciiLatin1Encoding): Hash;
+        update(data: string, input_encoding: Utf8AsciiLatin1Encoding): Hash;
         digest(): Buffer;
         digest(encoding: HexBase64Latin1Encoding): string;
     }
     export interface Hmac extends NodeJS.ReadWriteStream {
         update(data: string | ArrayBufferView): Hmac;
-        update(data: string | ArrayBufferView, input_encoding: Utf8AsciiLatin1Encoding): Hmac;
+        update(data: string, input_encoding: Utf8AsciiLatin1Encoding): Hmac;
         digest(): Buffer;
         digest(encoding: HexBase64Latin1Encoding): string;
     }
-
+    export type CipherAADTypes = 'aes-128-ccm' | 'aes-192-ccm' | 'aes-256-ccm' | 'aes-128-gcm' | 'aes-192-gcm' | 'aes-256-gcm'
+    export type CipherAADOptions extends stream.TransformOptions {
+        authTagLength: number;
+    }
     /** @deprecated since v10.0.0 use createCipheriv() */
     export function createCipher(algorithm: string, password: string | ArrayBufferView, options?: stream.TransformOptions): Cipher;
+    export function createCipher(algorithm: CipherAADTypes, password: string | ArrayBufferView, options: CipherAADOptions): CipherAAD;
+
     export function createCipheriv(algorithm: string, key: string | ArrayBufferView, iv: string | ArrayBufferView, options?: stream.TransformOptions): Cipher;
+    export function createCipheriv(algorithm: CipherAADTypes, key: string | ArrayBufferView, iv: string | ArrayBufferView, options: CipherAADOptions): CipherAAD;
+
     export interface Cipher extends NodeJS.ReadWriteStream {
         update(data: string | ArrayBufferView): Buffer;
-        update(data: string | ArrayBufferView, input_encoding: Utf8AsciiBinaryEncoding): Buffer;
-        update(data: ArrayBufferView, input_encoding: any, output_encoding: HexBase64BinaryEncoding): string;
+        update(data: string, input_encoding: Utf8AsciiBinaryEncoding): Buffer;
+        update(data: ArrayBufferView, input_encoding?: any, output_encoding: HexBase64BinaryEncoding): string;
```

Comment:
I don't think an optional before a mandatory arg is allowed.

Reply:
CI will tell once you fix the merge conflict.

Reply:
yep